staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -227,10 +199,34 @@ func (s *store) conditionalDelete(ctx context.Context, key string, out runtime.O
 	}
 	for {
 		origState, err := s.getState(getResp, key, v, false)
-		if err != nil {
+		if err != nil && !storage.IsTransformerError(err) {
 			return err
 		}
-		if err := preconditions.Check(key, origState.obj); err != nil {
+		if err != nil && storage.IsTransformerError(err) && preconditions != nil {
+			return err
+		}
+		if err != nil && storage.IsTransformerError(err) && preconditions == nil {
```

Comment:
This might be controversial.

Reply:
this is interesting. wouldn't malformed data in etcd also prevent us from listing and finding out what the problematic key was? I'm trying to understand if this special handling actually enables us to resolve the issue, or if it's just a step in that direction.

Reply:
Good point. The error message send by the LIST handler does include the key: https://github.com/kubernetes/kubernetes/blob/0889c3e4e9f3f95232301de5432df0e21b3ee2c1/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L610

So this special handling enables us to resolve the issue.